Async event action mapping to 5 min, add warning

mariusandra opened the pull request:
## Changes

- Users were maxing out their postgres CPUs
- Sets async event action mapping to 5min
- Adds an env `ACTION_EVENT_MAPPING_INTERVAL_SECONDS` to further specify the duration if needed
- Adds a warning describing the situation. Dismissing the warning dismisses it forever.
![image](https://user-images.githubusercontent.com/53387/111999199-0d657f00-8b1d-11eb-8aad-a16e9ed232b0.png)


## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

mariusandra committed:
Set ACTION_EVENT_MAPPING_INTERVAL_SECONDS to 300, move to settings

mariusandra committed:
Add async warning

timgl commented on frontend/src/layout/navigation/AsyncActionWarning.tsx:
```diff
@@ -0,0 +1,47 @@
+                message={'Webhooks and actions delayed up to 5 minutes'}
+                className="demo-warning"
+                description={
+                    <>Due to temporary limitations, webhooks and actions might be delayed by up to 5 minutes.</>
```
I don't think this will be temporary?

mariusandra replied:
Until we move action event mapping to the plugin server, as discussed here: https://github.com/PostHog/plugin-server/issues/235

Twixes replied:
But will action mapping in the plugin server by itself make things better? 🤔 Postgres is the bottleneck in this case and action mapping on the fly will (did?) in fact hammer it even more, because every event is a whole new request. I think what may need optimization is action mapping queries/data layout itself. Otherwise action matching without significant delays seems to be hard to pull off. Feel free to correct me

mariusandra replied:
Action event mapping in the plugin server wouldn't make extra queries, that's the thing :).

Except if a) needing to fetch cohorts to see if a person belongs to them due to a filter, or b) fetching a person to match person properties in a filter. The person is more often than not already fetched, and we could inline adding the cohorts with a subquery if needed.

timgl commented on frontend/src/scenes/App.tsx:
```diff
@@ -117,6 +118,7 @@ export function App(): JSX.Element | null {
                     {!sceneConfig.hideTopNav && <TopNavigation />}
                     {scene ? (
                         <Layout.Content className="main-app-content" data-attr="layout-content">
+                            {user.is_async_event_action_mapping_enabled && <AsyncActionWarning />}
```
I wasn't suggesting always showing this, just when you are creating a webhook (and maybe when you are looking at an action).

Twixes committed:
Use smarter get_from_env

mariusandra committed:
Async event action mapping warning when creating a webhook

mariusandra committed:
Async event action mapping warning when editing an action

mariusandra committed:
remove top async warning

mariusandra commented:
I removed the top banner and just added these lines:

![image](https://user-images.githubusercontent.com/53387/112001933-af866680-8b1f-11eb-9b20-61cd39697503.png)

![image](https://user-images.githubusercontent.com/53387/112001884-a72e2b80-8b1f-11eb-807d-1cc67f314c5f.png)

timgl commented:
yeah looks good

mariusandra commented:
WTF is this now... 🤔

![image](https://user-images.githubusercontent.com/53387/112002690-74d0fe00-8b20-11eb-8a75-acbca8bec8fd.png)

Considering I didn't touch that file and want to cherry-pick this into a patch, I won't touch that line and will just merge once everything else is green.

Twixes commented:
It was broken in #3666.

Twixes committed:
Use common reworded notice component for async mapping

Twixes commented on frontend/src/scenes/project/Settings/WebhookIntegration.tsx:
```diff
@@ -114,6 +114,15 @@ const logic = kea<logicType<UserType>>({
 
+export function AsyncActionMappingNotice(): JSX.Element {
+    return (
+        <p>
+            Please note that webhooks and actions may be delayed up to 5 minutes due to open-source PostHog
+            configuration.
+        </p>
```
I committed this notice. I'd really like to be clear for users so that they get why there's a delay, WDYT here @mariusandra?

mariusandra replied:
Good enough for me

mariusandra committed:
Merge branch 'master' into async-warning-300
